acceptance/tests/parser_functions/hiera_in_templates.rb

```diff
@@ -0,0 +1,403 @@
+  #    │                   └── hieratest_results_erb.erb
+  #    ├── hieradata
+  #    │   ├── default.yaml
+  #    │   ├── production.yaml
+  #    │   └── RedHat.yaml
+  #    ├── hiera.yaml
+  #    └── puppet.conf
+
+
+
+  envroot = "#{$coderoot}/environments"
+  production = "#{envroot}/production"
+  modroot = "#{production}/modules"
+  moduledir = "#{modroot}/#{$module_name}"
+  hieradir = "#{$coderoot}/hieradata"
+
+  # {
+  environ = <<ENV
+
+File {
+  ensure => file,
+  owner => #{master.puppet['user']},
+  group => #{master.puppet['group']},
+  mode   => "0644",
+}
+
+file {
+  [
+    "#{$coderoot}",
+    "#{envroot}",
+    "#{production}",
+    "#{production}/modules",
+    "#{production}/manifests",
+    "#{hieradir}",
+    "#{moduledir}",
+    "#{moduledir}/examples",
+    "#{moduledir}/manifests",
+    "#{moduledir}/templates",
+  ] :
+  ensure => directory,
+}
+
+file { '#{production}/manifests/site.pp':
+  ensure => file,
+  content => "
+node default {
+  \\$msgs = hiera_array('message')
+  notify {\\$msgs:}
+  include #{$module_name}
+}
+",
+}
+
+
+file {"#{$coderoot}/hiera.yaml":
+  content => "
+---
+:backends:
+  - yaml
+
+:yaml:
+  :datadir: #{$coderoot}/hieradata
+
+:hierarchy:
+  - \\"%{clientcert}\\"
+  - \\"%{environment}\\"
+  - \\"%{osfamily}\\"
+  - \\"default\\"
+"
+}
+
+file {"#{hieradir}/default.yaml":
+  content => "
+---
+message: '#{$msg_default}'
+includes: '#{$module_name}::mod_default'
+"
+}
+
+file {"#{hieradir}/${osfamily}.yaml":
+  content => "
+---
+message: [
+  '#{$msg1os}',
+  '#{$msg2os}',
+]
+includes: '#{$module_name}::mod_osfamily'
+hash_value:
+  #{$k3}: '#{$hval3os}'
+  #{$k2}: '#{$hval2os}'
+"
+}
+
+file {"#{hieradir}/production.yaml":
+  content => "
+---
+message: '#{$msg_production}'
+includes: '#{$module_name}::mod_production'
+hash_value:
+  #{$k1}: '#{$hval1p}'
+  #{$k2}: '#{$hval2p}'
+"
+}
+
+file {"#{$hieradir}/#{$fqdn}.yaml":
+  content => "
+---
+message: '#{$msg_fqdn}'
+includes: '#{$module_name}::mod_fqdn'
+"
+}
+
+file {"#{moduledir}/examples/init.pp":
+  content => "
+include #{$module_name}
+"
+}
+
+file { "#{moduledir}/manifests/init.pp":
+  content => "
+class #{$module_name} {
+  file {
+    [
+      '#{$coderoot}',
+      '#{$resultdir}',
+    ]:
+    ensure => directory,
+    owner  => 'root',
+    group  => 'root',
+    mode   => '0755',
+  }
+  file {'/#{$resultdir}/#{$module_name}_results_epp':
+    ensure  => file,
+    owner => 'root',
+    group => 'root',
+    mode  => '0644',
+    content => epp('#{$module_name}/hieratest_results_epp.epp'),
+  }
+  file {'/#{$resultdir}/#{$module_name}_results_erb':
+    ensure  => file,
+    owner => 'root',
+    group => 'root',
+    mode  => '0644',
+    content => template('#{$module_name}/hieratest_results_erb.erb'),
+  }
+}
+"
+}
+
+file { "#{moduledir}/manifests/mod_default.pp":
+  content => "
+class #{$module_name}::mod_default {
+  notify{\\"module mod_default invoked.\\\\n\\":}
```

Comment:
you could use single quotes here rather than escaping the doubles.  also, in most cases the \n is implied in a notify's output.